.github/workflows/platform.librarycheck.psrule.yml

```diff
@@ -0,0 +1,108 @@
+
+            ## Add local (source control) tokens
+            $tokenMap = @{}
+            foreach ($token in (Get-ChildItem env: | Where-Object -Property Name -Like "localToken_*")) {
+              $tokenMap += @{ $token.Name.Replace('localToken_','','OrdinalIgnoreCase') = $token.value }
+            }
+            Write-Verbose ('Using local tokens [{0}]' -f ($tokenMap.Keys -join ', ')) -Verbose
+            $Tokens += $tokenMap
+
+            ## Swap 'namePrefix' token if empty and provided as a GitHub secret
+            if([String]::IsNullOrEmpty($Tokens['namePrefix'])){
+              Write-Verbose 'Using [namePrefix] token from GitHub' -Verbose
+              $Tokens['namePrefix'] = '${{ env.TOKEN_NAMEPREFIX }}'
+            }
+
+            # Get File Path List
+            $modulesFolderPath = Join-Path $env:GITHUB_WORKSPACE '${{ env.modulesPath }}'
+            $moduleTestFiles = [System.Collections.ArrayList]@()
+            $moduleTestFiles += Get-ChildItem -Path $env:GITHUB_WORKSPACE -Filter *.test.bicep -Recurse -Force -Name
+
+            # Construct Token Function Input
+            $ConvertTokensInputs = @{
+              FilePathList = $moduleTestFiles
+              Tokens       = $Tokens
+              TokenPrefix  = '${{ env.tokenPrefix }}'
+              TokenSuffix  = '${{ env.tokenSuffix }}'
+            }
+
+            Write-Verbose "Convert Tokens Input:`n $($ConvertTokensInputs | ConvertTo-Json -Depth 10)" -Verbose
+
+            # Invoke Token Replacement Functionality [For Module]
+            $null = Convert-TokensInFileList @ConvertTokensInputs -verbose
+
+            Write-Output '::endgroup::'
+      - name: Run PSRule analysis
+        uses: microsoft/ps-rule@v2.4.0
+        continue-on-error: true # Setting this whilst PSRule gets bedded in, in this project
+        with:
+          modules: 'PSRule.Rules.Azure'
+          inputPath: '${{ env.modulesPath }}/'
+          outputFormat: Csv
+          outputPath: '${{ env.modulesPath }}/PSRule-output.csv'
+      - name: 'Parse CSV content'
+        uses: azure/powershell@v1
+        with:
+          azPSVersion: 'latest'
+          inlineScript: |
+            # Grouping task logs
+            Write-Output '::group::Parse CSV content'
+
+            # Load used functions
+            . (Join-Path $env:GITHUB_WORKSPACE 'utilities' 'pipelines' 'PSRuleValidation' 'Set-PSRuleOutput.ps1')
+
+            # Populate parameter input
+            $ParameterInput = @{
+              inputFilePath           = '${{ env.modulesPath }}/PSRule-output.csv'
+              outputFilePath          = '${{ env.modulesPath }}/PSRule-output.md'
+              skipPassedRulesReport   = $true
+            }
+
+            # Invoke function
+            $null = Set-PSRuleOutput @ParameterInput
+
+            Write-Output '::endgroup::'
+      - name: Output to GitHub job summaries
+        if: always()
+        run: cat '${{ env.modulesPath }}/PSRule-output.md' >> $GITHUB_STEP_SUMMARY
```

Comment:
Question: Could we use that to publish our Pester test results too (in a separate PR, of course)? Like in a tab below? Right now we don't publishing anything but this small box that shows the number of failed/passed tests

Reply:
We can definitely print to job summaries for all GitHub actions. I agree using that with Pester could improve the troubleshooting as well. I wouldn't mix the two though, but instead get rid of the external action we're currently using to publish Pester tests and implement something on our end.

Reply:
Completely agree. It should be one or the other. I'll create a new issue to implement this and follow your example here. Great work :)

Reply:
Created issue: https://github.com/Azure/ResourceModules/issues/2575